x-pack/plugins/fleet/server/services/security/security.ts

```diff
@@ -115,15 +131,56 @@ export async function getAuthzFromRequest(req: KibanaRequest): Promise<FleetAuth
       kibanaPrivileges: privileges.kibana,
       searchPrivilege: 'fleet-setup',
     });
+    // Fleet sub features
+    const fleetAgentsAllAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-agents-all`,
+    });
+    const fleetAgentsReadAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-agents-read`,
+    });
+    const fleetAgentPoliciesAllAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-agent-policies-all`,
+    });
+    const fleetAgentPoliciesReadAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-agent-policies-read`,
+    });
+    const fleetSettingsAllAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-settings-all`,
+    });
+    const fleetSettingsReadAuth = getAuthorizationFromPrivileges({
+      kibanaPrivileges: privileges.kibana,
+      prefix: `${PLUGIN_ID}-settings-read`,
+    });
 
     return {
       ...calculateAuthz({
-        fleet: { all: fleetAllAuth, setup: fleetSetupAuth },
+        fleet: {
+          all: fleetAllAuth,
+          read: fleetReadAuth,
+          setup: fleetSetupAuth,
+          agents: {
+            read: fleetAgentsReadAuth,
+            all: fleetAgentsAllAuth,
+          },
+          agentPolicies: {
+            read: fleetAgentPoliciesAllAuth,
+            all: fleetAgentPoliciesReadAuth,
+          },
+          settings: {
+            read: fleetSettingsReadAuth,
+            all: fleetSettingsAllAuth,
+          },
+        },
         integrations: {
           all: intAllAuth,
           read: intReadAuth,
         },
-        isSuperuser: checkSuperuser(req),
```

Comment:
is the superuser not used anymore? is there any impact when we are migrating kibana?

Reply:
No it's not used anymore since https://github.com/elastic/kibana/pull/178120 just doing some cleanup